docker-entrypoint.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+export CONFIG_DIR="${CONFIG_DIR:-/opt/config}"
+export CONFIG=$CONFIG_DIR/config.json
+
+mkdir -p $CONFIG_DIR
+chmod 777 -R $CONFIG_DIR
+
+[[ -f $CONFIG_DIR/package.json ]] || cp install/package.json $CONFIG_DIR/package.json
+[[ -f $CONFIG_DIR/package-lock.json ]] || touch $CONFIG_DIR/package-lock.json
+
+ln -s $CONFIG_DIR/package.json package.json
+ln -s $CONFIG_DIR/package-lock.json package-lock.json
+
+npm install --only=prod
+
+if [ -f $CONFIG ]; then
+  echo "Config file exist at $CONFIG, assuming it is a valid config"
+  echo "Starting forum"
+  ./nodebb build --config=$CONFIG
```

Comment:
Is there a reason to use (or at least expose in docker-compose) a volume for build results if NodeBB is supposed be rebuilt on every start anyway?

Reply:
Long startup times of containers are a semi-fundamental issues. E.g. Java apps get a lot of hatred for this. But nodebb is even worse with building the whole thing at startup instead of at container build time. Storing the build in a volume prevents such long startups at the cost of manual rebuild if things have changed.
I'm running a similar setup for quite a while. It has its disadvantages too.

Reply:
Yeah, my issue is that it seems redundant if it's being built every time anyway; I'm fairly sure `nodebb build` doesn't use any cache, so every start means a full build.
So I think either it shouldn't be built at startup with the build results stored in a container, or the volume could be removed if that build is desirable.

Reply:
Yes, I understand why building on startup was chosen, what I don't understand is storing the build in a volume then.

The only instance it changes something is if one overrides the enrypoint in order to bypass build process, which I guess may be enough reason to leave that in.

Reply:
Ah, seems it's handled with an environmental variable now.